xcpmd/src/xcpmd-dbus-server.c

```diff
@@ -440,20 +437,7 @@ gboolean xcpmd_battery_percentage(XcpmdObject *this, guint IN_bat_n, guint *OUT_
         return FALSE;
     }
 
-    juice_left = last_status[IN_bat_n].remaining_capacity;
-    juice_when_full = last_info[IN_bat_n].last_full_capacity;
-
-    /* If there's no last_full_capacity, try design_capacity */
-    if (juice_when_full == 0)
-        juice_when_full = last_info[IN_bat_n].design_capacity;
-
-    /* Let's not divide by 0 */
-    if (juice_when_full == 0) {
-        g_set_error(error, DBUS_GERROR, DBUS_GERROR_FAILED, "Unhappy battery: %d", IN_bat_n);
-        return FALSE;
-    }
-
-    *OUT_percentage = juice_left * 100 / juice_when_full;
+    *OUT_percentage = get_battery_percentage(IN_bat_n);
```

Comment:
A whole bunch of code is removed here and replaced with a call to get_battery_percentage(). I would expect that function to be defined in this PR. I guess I am confused about this.